<?xml version="1.0"?>
<launch>

  <arg name="cf_num" default="1" />
  <arg name="z_coefficient" default="4"/>
  <arg name="continuous" default="True" />
  <arg name="air_vel" default="1.0" />
  <arg name="land_vel" default=".3" />
  <arg name="follow" default="False" />
  <arg name="follow_nodes" default="3cf_2steps_everytile" />

  <remap from="opt_sg/StartingID_topic" to="setup/StartingID_topic" />
  <remap from="opt_planner/Starter" to="setup/Starter" />

  <node name="opt_planner" pkg="planner" type="opt_planner.py" output="screen">
    <param name="cf_num" value="$(arg cf_num)"/>
    <param name="z_coefficient" value="$(arg z_coefficient)" />
    <param name="continuous" value="$(arg continuous)" />
    <param name="land_vel" value="$(arg land_vel)" />
    <param name="air_vel" value="$(arg air_vel)" />
    <param name="air_buffer_dist" value=".5" />
  </node>

  <node name="opt_sg" pkg="planner" type="opt_sg.py" >
    <param name="follow" value="$(arg follow)" />
    <rosparam command="load" file="$(find planner)/config/$(arg follow_nodes).yaml" />
  </node>

  <node name="setup" pkg="planner" type="setup.py" output="screen">
    <param name="cf_num" value="$(arg cf_num)" />
    <param name="centered" value="True" />
    <param name="follow" value="$(arg follow)" />
    <rosparam command="load" file="$(find planner)/config/$(arg follow_nodes).yaml" />
  </node>

</launch>